File: melhor-consulta-spring/src/main/java/com/soaresdev/melhorconsultaspring/models/Medico.java
package com.soaresdev.melhorconsultaspring.models;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Data
public class Medico{

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nomeSocial;
    private String nome;
    private String sobrenome;
    private String cpf;
    private String email;
    private String celular;
    private String sexo;
    private Date dtNascimento;
    private String logradouro;
    private String nroLogradouro;
    private String cep;
    private String cidade;
    private String estado;
    private String cnpj;
    private String crm;
    private Date dtInicio;
    @OneToMany
    private List<GradeConsulta> gradeConsultas;
    @OneToMany(mappedBy = "medico")
    private List<Especialidade> especialidades;
    private boolean situacao;
    private Integer notaAvaliacao;
}

File: melhor-consulta-angular/src/app/models/medico.ts
import { Especialidade } from './especialidade';
import { GradeConsulta } from './gradeConsulta';
export interface Medico {
  id: number,
  nome: string,
  sobrenome: string,
  nomeSocial: string,
  email: string,
  cpf: string,
  cnpj: string,
  crm: string,
  dtInicio: string,
  dtNascimento: string,
  logradouro: string,
  nroLogradouro: string,
  cep: string,
  estado: string,
  grades: GradeConsulta[],
  especialidades: Especialidade[],
  situacao: boolean,
  notaAvaliacao: number
}

export interface MedicoRequest{
  nome: string,
  sobrenome: string,
  senha: string,
  nomeSocial: string,
  email: string,
  cpf: string,
  crm: string,
  dtNascimento: string,
  logradouro: string,
  complemento: string,
  nroLogradouro: string,
  bairro: string,
  localicade: string,
  cep: string,
  uf: string,
  especialidades: number[],
}
